JSONModel/JSONModel/JSONModel.m

```diff
@@ -721,6 +718,29 @@ -(void)__inspectProperties
                              );
 }
 
+-(void)__addCustomSetterForName:(NSString*) name
+                     withSuffix:(NSString*) suffix
+                            key:(NSString*) key
+                     toProperty:(JSONModelClassProperty*) p{
+    if (p.customSetters[key])
+        return;
+    
+    SEL setter = NSSelectorFromString([NSString stringWithFormat:@"set%@With%@:", name, suffix]);
+    
+    if ([self respondsToSelector:setter]){
+        NSValue* setterValue = [NSValue valueWithBytes:&setter objCType:@encode(SEL)];
+        p.customSetters[key] = [[JSONModelCustomSetter alloc] initWithValue:setterValue withErrorOutParam:NO];
+    }
+
+    SEL setterWithError = NSSelectorFromString([NSString stringWithFormat:@"set%@With%@:error:", name, suffix]);
```

Comment:
Check for the error setter first, then return within the `respondsToSelector` conditional block